src/udf/js_to_si.c

```
@@ -0,0 +1,276 @@
+			if (JS_IsArray (js_ctx, val)) {
+				ret = _JSArrayToSIValue (js_ctx, val) ;
+			} else if (JS_IsObject (val)) {
+				// handle registered Falkor classes
+				JSClassID cid = JS_GetClassID (val) ;
+
+				if (cid == js_node_class_id) {
+					Node *n = JS_GetOpaque (val, js_node_class_id) ;
+
+					// clone node
+					Node *_n = rm_malloc (sizeof (Node)) ;
+					memcpy (_n, n, sizeof (Node)) ;
```

Cppcheck (2.18.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)